scripts/build.ps1

```diff
@@ -170,6 +170,37 @@ function Invoke-TestAssetsBuild
         Write-Log ".. .. Build: Source: $TPB_TestAssets_Solution -- add NuGet source"
         Invoke-Exe -IgnoreExitCode 1 $nugetExe -Arguments "sources add -Name ""locally-built-testplatform-packages"" -Source $env:TP_TESTARTIFACTS\packages\ -ConfigFile ""$nugetConfig"""
         Invoke-Exe $dotnetExe -Arguments "build $TPB_TestAssets_Solution --configuration $TPB_Configuration -v:minimal -p:CIBuild=$TPB_CIBuild -p:LocalizedBuild=$TPB_LocalizedBuild -bl:""$env:TP_OUT_DIR\log\$Configuration\TestAssets.binlog"""
+
+        $dependenciesPath = "$env:TP_ROOT_DIR\scripts\build\TestPlatform.Dependencies.props"
+        $dependenciesXml = [xml](Get-Content -Raw -Encoding UTF8 $dependenciesPath)
+
+        $projects = @(
+            "$env:TP_ROOT_DIR\test\TestAssets\SimpleTestProject\SimpleTestProject.csproj"
+            "$env:TP_ROOT_DIR\test\TestAssets\SimpleTestProject2\SimpleTestProject2.csproj"
+        )
+
+        $versionProperties = @(
+            "MSTestFrameworkLatestStableVersion" 
+            "MSTestFrameworkLatestPreviewVersion"
+            "MSTestFrameworkMostDownloadedVersion"
+            "MSTestFrameworkPreviousStableVersion"
+            "MSTestFrameworkLegacyStableVersion"
+        )
+        
+        foreach ($project in $projects) {
+            foreach ($propertyName in $versionProperties) {             
+                $mstestVersion = $dependenciesXml.Project.PropertyGroup.$propertyName
+
+                if (-not $mstestVersion)
+                {
+                    throw "MSTestVersion for $propertyName is empty."
+                }
+                
+                $dirVersion = $mstestVersion  -replace "\[|\]" 
+                $dirPropertyName = $propertyName -replace "Framework" -replace "Version"
+                Invoke-Exe $dotnetExe -Arguments "build $project --configuration $TPB_Configuration -v:minimal -p:CIBuild=$TPB_CIBuild -p:LocalizedBuild=$TPB_LocalizedBuild -p:MSTestFrameworkVersion=$mstestVersion -p:MSTestAdapterVersion=$mstestVersion -p:BaseOutputPath=""bin\$dirPropertyName-$dirVersion\\""" 
+            }
```

Comment:
I pre-build all the dlls that we use to run acceptance tests with their respective versions of mstest (and later xunit and nunit).